src/net.cpp

```diff
@@ -960,12 +960,18 @@ static bool AttemptToEvictConnection(bool fPreferNewConnection)
         static int64_t nLastTime = GetTime();
         for (CNode *node : vNodes)
         {
-            // Decay the activity bytes for each node over a period of 2 hours.  This gradually de-prioritizes a
-            // connection
-            // that was once active but has gone stale for some reason and allows lower priority active nodes to climb
-            // the ladder.
+            // Decay the activity bytes for each node over a period of 2 hours.  This gradually de-prioritizes
+            // a connection that was once active but has gone stale for some reason and allows lower priority
+            // active nodes to climb the ladder.
             int64_t nNow = GetTime();
-            node->nActivityBytes *= pow(1.0 - 1.0 / 7200, (double)(nNow - nLastTime)); // exponential 2 hour decay
+
+            while (true)
+            {
+                uint64_t nOldActivityBytes = node->nActivityBytes;
+                uint64_t nNewActivityBytes = nOldActivityBytes * pow(1.0 - 1.0 / 7200, (double)(nNow - nLastTime));
+                if (node->nActivityBytes.compare_exchange_weak(nNewActivityBytes, nOldActivityBytes))
```

Comment:
https://en.cppreference.com/w/cpp/atomic/atomic/compare_exchange:
bool compare_exchange_weak( T& expected, T desired,
                            std::memory_order success,
                            std::memory_order failure )

do you have compare_exchange_weak(nNewActivityBytes, nOldActivityBytes) backwards?

Reply:
yes you're right, thx...fixed.